source/world_builder/utilities.cc

```diff
@@ -1346,6 +1348,27 @@ namespace WorldBuilder
           const double spreading_velocity_point0 = mid_oceanic_spreading_velocities[relevant_ridge][i_coordinate];
           const double spreading_velocity_point1 = mid_oceanic_spreading_velocities[relevant_ridge][i_coordinate + 1];
 
+          // When subducting_velocities is not input by the user, default value is 0, which
+          // results in subducting velocity == spreading_velocity. When a single value is
+          // input by the user, subducting velocity != spreading_velocity, but
+          // subducting velocity is spatially constant.
+          double subducting_velocity_point0 = subducting_plate_velocities[0][0];
+          double subducting_velocity_point1 = subducting_plate_velocities[0][0];
+
+          // When subducting_velocities is input as an array, spatial variation
+          if (subducting_plate_velocities[0].size() > 1)
+            {
+              WBAssertThrow(subducting_plate_velocities.size() == mid_oceanic_ridges.size() && \
+                            subducting_plate_velocities[relevant_ridge].size() == mid_oceanic_ridges[relevant_ridge].size(),
+                            "subducting velocity and ridge coordinates must be the same dimension");
+              WBAssertThrow(ridge_migration_times.size() == mid_oceanic_ridges.size(),
```

Comment:
As discussed, copy this to parse parameters, and turn these into WBAssert (which is only active in debug mode)